http4s-scala-fx/src/test/scala/http4s/fx/HttpServerFixtures.scala

```diff
@@ -0,0 +1,38 @@
+package http4s
+package fx
+
+import _root_.fx.{handle, structured, toEither, Control, Resource}
+import _root_.fx.instances.FxAsync.asyncInstance
+import _root_.fx.instances.StructuredF
+import _root_.fx.HttpExecutionException
+import fx.BlazeServerFXBackend
+import munit.FunSuite
+import org.http4s.Method.{GET, POST}
+import org.http4s.{HttpApp, HttpRoutes, Response}
+
+trait HttpServerFixtures:
+  self: FunSuite =>
+
+  val pingPongApp: HttpApp[StructuredF] =
+    HttpApp.apply {
+      case r if r.method == GET && r.uri.path.renderString.contains("ping") =>
+        Response[StructuredF]().withEntity[String]("pong")
+    }
+
+  val echoApp: HttpApp[StructuredF] =
+    HttpApp.apply {
+      case r if r.method == POST && r.uri.path.renderString.contains("echo") =>
+        Response[StructuredF]().withBodyStream(r.body)
+    }
+
+  def httpServerHttp4s(app: HttpApp[StructuredF]): FunFixture[Resource[String]] =
```

Comment:
```suggestion
  def httpServerHttp4s(app: HttpApp[StructuredF]): FunFixture[Resource[(Structured, Control[Throwable]) ?=> String]] =
```
This guarantees that the resource returned will have a `Structured` for fixture `setup`, and that the continuation will have a control token to handle the failure in the tests.